lib/at_cmd/at_cmd.c

```diff
@@ -136,24 +203,25 @@ static void socket_thread_fn(void *arg1, void *arg2, void *arg3)
 	LOG_DBG("AT socket thread started");
 
 	for (;;) {
-		LOG_DBG("Allocating memory slab for AT socket");
-		k_mem_slab_alloc(&rsp_work_items, (void **)&item, K_FOREVER);
-		LOG_DBG("Allocation done");
+		LOG_DBG("Writing any pending command");
+		load_cmd_and_write();
```

Comment:
I a bit confused why you need this, the queue is already processed when you issue the command. I can not see any way to insert commands in the queue without them beeing processed righ away.

Reply:
My answer on 307 should clarify this :)